### ISSUE TITLE
key encryption of cert tars

### ISSUE THREAD

**Issue by ianmiell** [IshentRas/cookbook-openshift3#161]
We are going to propose a change to allow the encryption/decryption of the cert tars:

```
openssl enc -aes-256-cbc -in "UNENCRYPTEDFILEVAL" -out "ENCRYPTEDFILEVAL" -k "PASSVAL"
openssl enc -d -aes-256-cbc -in "ENCRYPTEDFILEVAL" -out "UNENCRYPTEDFILEVAL" -k "PASSVAL"
```

The intention is to wrap this in a flag, and by default nothing will change.

Code and PR to follow.

**Comment by ianmiell** [comment]
Some work done on this.

I've changed the idea slightly: _all_ tgz's will be encrypted with a password, that is set in the attributes/default.rb with a default.

Those that want to override this with a encrypted data bag can ensure that that value is set appropriately. Testing.

**Comment by ianmiell** [comment]
This situation sometimes occurs:

```
[root@etcd1 etcd]# ls
ca.crt  etcd.conf  etcd-etcd1.vagrant.test.tgz  etcd-etcd1.vagrant.test.tgz.enc  peer.crt  peer.key  server.crt  server.key
[root@etcd1 etcd]# ls -lrt
total 28
-rw-r--r--. 1 root root 10688 Sep 29 15:08 etcd-etcd1.vagrant.test.tgz.enc
-rw-------. 1 etcd etcd     0 Sep 29 15:09 ca.crt
-rw-------. 1 etcd etcd     0 Sep 29 15:09 server.crt
-rw-------. 1 etcd etcd     0 Sep 29 15:09 server.key
-rw-------. 1 etcd etcd     0 Sep 29 15:09 peer.crt
-rw-------. 1 etcd etcd     0 Sep 29 15:09 peer.key
-rw-r--r--. 1 root root   998 Sep 29 15:09 etcd.conf
-rw-r--r--. 1 root root 10662 Sep 29 15:51 etcd-etcd1.vagrant.test.tgz
[root@etcd1 etcd]# tar -zxvf etcd-etcd1.vagrant.test.tgz
./
./server.key
./server.csr
./server.crt
./peer.key
./peer.csr
./peer.crt
./ca.crt
[root@etcd1 etcd]# ls -lrt
total 64
-rw-r--r--. 1 root root  1704 Sep 29 14:59 server.key
-rw-r--r--. 1 root root   985 Sep 29 14:59 server.csr
-rw-r--r--. 1 root root  5832 Sep 29 14:59 server.crt
-rw-r--r--. 1 root root  1704 Sep 29 14:59 peer.key
-rw-r--r--. 1 root root   985 Sep 29 14:59 peer.csr
-rw-r--r--. 1 root root  5879 Sep 29 14:59 peer.crt
-rw-r--r--. 1 root root  1895 Sep 29 14:59 ca.crt
-rw-r--r--. 1 root root 10688 Sep 29 15:08 etcd-etcd1.vagrant.test.tgz.enc
-rw-r--r--. 1 root root   998 Sep 29 15:09 etcd.conf
-rw-r--r--. 1 root root 10662 Sep 29 15:51 etcd-etcd1.vagrant.test.tgz
[root@etcd1 etcd]# systemctl start etcd
```

and I'm not sure why. I might add a check to remove the tgz and the empty files if they are there, but this is obviously just a workaround.

**Comment by ianmiell** [comment]
Figured it out - it's to do with ordering of events. Corrected the notifies entries in etcd.

**Comment by IshentRas** [comment]
Fixed #163
